lib/link.js

```diff
@@ -100,11 +107,13 @@ Link.prototype.attach = function() {
 
 Link.prototype.detach = function() {
   var self = this;
+  this._timeouts = undefined; // Disable any re-attachment policy.
   var detachPromise = new Promise(function(resolve, reject) {
     var onError = function(err) { reject(err); };
     self.once(Link.ErrorReceived, onError);
     self.once(Link.Detached, function(info) {
       self.removeListener(Link.ErrorReceived, onError);
+      self.session._removeLink(self);
```

Comment:
No, it removes it _only_ when the detach comes from our side. When the detach is instigated by the server, it leaves it alone.